map/bookmark_manager.cpp

```diff
@@ -1864,28 +1864,19 @@ void BookmarkManager::LoadBookmarkRoutine(std::string const & filePath, bool isT
 
     auto collection = std::make_shared<KMLDataCollection>();
 
+    // Convert KML/KMZ/KMB files to temp KML file.
     std::string fileSavePath = GetKMLPath(filePath);
     if (!fileSavePath.empty())
     {
-      auto const ext = GetLowercaseFileExt(filePath);
       std::unique_ptr<kml::FileData> kmlData;
-      if (ext == ".kml" || ext == ".kmz")
```

Comment:
Are you 100% sure that this code should be removed? Did you test it with other file types? Is it guaranteed that this code is called only for kml/kmz files?

Reply:
I tested with KML, KMZ, KMB, and GPX files.
Found an issue with GPX and fixed. Current implementation imports all formats successfully.

One questionable part: KMB is first converted to temp KML and then KML is converted to final KML. Maybe we can skip temporary step.

Nice to have: unit tests 🙂

Reply:
Converting kml to kml doesn't look necessary ) Is it easy to fix?